src/Entity.php

```diff
@@ -20,9 +20,36 @@ public function getContentType() {
 	}
 
 	public function bind($entityType) {
-		if (!class_exists($entityType) || !in_array('Fliglio\Web\MappableApi', class_implements($entityType))) {
-			throw new \Exception($entityType . " doesn't implement Fliglio\Web\MappableApi");
+		$this->checkType($entityType);
+
+		$arr = $this->parseBody();
+
+		$entity = $entityType::unmarshal($arr);
+
+		if ($entity instanceof Validation) {
+			// throws ValidationException on constraint violation
+			$entity->validate();
 		}
+		return $entity;
+	}
+
+	public function bindCollection($entityType) {
+		$this->checkType($entityType);
+
+		$arr = $this->parseBody();
+
+		$mapper = new CollectionApiMapper($entityType::getApiMapper());
+		$coll = $mapper->unmarshal($arr);
+		foreach ($coll as $entity) {
+			if ($entity instanceof Validation) {
```

Comment:
Is it possible that any of the mappers could have validation objects more than 1 deep?

Reply:
This is operating on a collection of objects which is the output of the CollectionApiMapper::unmarshal.  e.g. 
```
[Thing1, Thing2, Thing3, ..., ThingN] 
```

If those objects (e.g. Thing) has subcomponents, Thing should trigger that in it's Assert/Valid on them.